jena-arq/src/main/java/org/apache/jena/riot/out/JenaRDF2JSONLD.java

```diff
@@ -0,0 +1,100 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.jena.riot.out;
+
+import java.util.Iterator ;
+
+import org.apache.jena.atlas.logging.Log ;
+import org.apache.jena.datatypes.xsd.XSDDatatype ;
+import org.apache.jena.graph.Node ;
+import org.apache.jena.riot.RiotException ;
+import org.apache.jena.riot.system.SyntaxLabels ;
+import org.apache.jena.sparql.core.DatasetGraph ;
+import org.apache.jena.sparql.core.Quad ;
+
+import com.github.jsonldjava.core.JsonLdError ;
+import com.github.jsonldjava.core.RDFDataset ;
+
+/** Convert from JSON-LD syntax to JSON-LD internal representation of a dataset, using Jena Quads/Nodes etc */
+class JenaRDF2JSONLD implements com.github.jsonldjava.core.RDFParser {
+    NodeToLabel labels = SyntaxLabels.createNodeToLabel() ;
+
+    @Override
+    public RDFDataset parse(Object object) throws JsonLdError {
+        RDFDataset result = new RDFDataset() ;
+        if ( object instanceof DatasetGraph )
+        {
+            DatasetGraph dsg = (DatasetGraph)object ;
+
+            Iterator<Quad> iter = dsg.find() ;
+            for ( ; iter.hasNext() ; )
```

Comment:
Converting loops to `forEachRemaining` is personal taste.

If a loop can be replaces by steram actions, then one thing, but this is turning a loop into a different loop!

`forEachRemaining` is reported to be slower. That might be just that the optimizer has not caught up but something needs to account for the needs of stack frames in case of an exception and that means extra work must be done.